### ISSUE TITLE
"No credentials provided for build!!!" since v1.8.0

### ISSUE THREAD

**Issue by lvdh** [jenkinsci/aws-beanstalk-publisher-plugin#51]
Hi all,

Shortly after upgrading the aws-beanstalk-publisher-plugin on our Jenkins masters (v1.7.4 to v1.8.0), we noticed a build failing with the `No credentials provided for build!!!` error message.

After rolling the plugin back to v1.7.4, a new build of the affected job successfully completed.

Build results snippet for **v1.7.4**:
```
+ zip -9 -q -r application.zip .
Root File Object is a file. We assume its a zip file, which is okay.
Uploading file awseb-[...].zip as s3://[...].zip
Upload took 8.260 s
Creating application version frontend-dev1-162 for application frontend for path s3://[...].zip
Environment found (environment id='[...]', name='frontend-dev1'). Attempting to update environment to version label 'frontend-dev1-162'
```

Build results snippet for **v1.8.0**:
```
+ zip -9 -q -r application.zip .
No credentials provided for build!!!
No credentials provided for build!!!
Root File Object is a file. We assume its a zip file, which is okay.
Uploading file awseb-[...].zip as s3://[...].zip
ERROR: Build step failed with exception
com.amazonaws.services.s3.model.AmazonS3Exception: Access Denied (Service: Amazon S3; Status Code: 403; Error Code: AccessDenied; Request ID: [...]
```

Going through the [comparison of both release commits](https://github.com/DavidTanner/aws-beanstalk-publisher/compare/510eddf...c8c6ed7), I could not find a change that would cause this issue. To be fair, I'm not at all familiar with the source code, it has been a while since I wrote any Java as well.

AWS credentials are configured in "Configure System" --> "Deploy into AWS Elastic Beanstalk".

Are we missing something? Perhaps we should adapt our job/Jenkins configuration to the new version of the plugin?

Thanks in advance for reading this far.

**Comment by jstenquist** [comment]
Is there a way to resolve this besides downgrading to 1.7.x.?

**Comment by DavidTanner** [comment]
Should be fixed in 1.8.1

**Comment by lvdh** [comment]
Thanks, David!
